app/src/main/java/com/github/se/signify/ui/screens/tutorial/TutorialScreen.kt

```diff
@@ -30,158 +20,191 @@ import androidx.compose.ui.graphics.Color
+                            y = feedbackPos.y,
+                            width = feedbackPos.width,
+                            height = feedbackPos.height),
+                        overlayColor)
+                  } else {
+                    HighlightedArea.FullScreenOverlayArea(overlayColor)
+                  },
+              textTag = "ExerciseTextTag"),
+          TutorialStep(
+              text = stringResource(R.string.tutorial_completionText),
+              highlightArea = HighlightedArea.FullScreenOverlayArea(overlayColor),
+              textTag = "CompletionTextTag"))
+
   Box(modifier = Modifier.fillMaxSize()) {
     // Display the HomeScreen in the background
     HomeScreen(navigationActions)
 
-    // Add an overlay to block input and gestures
-    Box(
-        modifier =
-            Modifier.fillMaxSize().background(Color.Transparent).clickable(
-                indication = null,
-                interactionSource =
-                    remember {
-                      MutableInteractionSource()
-                    }) { /* Block all interactions, including scrolling */})
-
-    // Overlay for tutorial steps
-    when (step) {
-      0 ->
-          TutorialOverlay(
-              text = stringResource(R.string.tutorial_screen1_text),
-              highlightArea = { FullScreenSemiTransparentOverlay() },
-              onNext = { step++ },
-              textTag = "WelcomeTextTag")
-      1 ->
-          TutorialOverlay(
-              text = stringResource(R.string.tutorial_screen2_text),
-              highlightArea = {
-                HighlightedBox(
-                    xOffset = 90.dp,
-                    yOffset = 280.dp,
-                    highlightWidth = 230.dp,
-                    highlightHeight = 190.dp)
-              },
-              onNext = { step++ },
-              textTag = "ExerciseTextTag")
-      2 ->
-          TutorialOverlay(
-              text = stringResource(R.string.tutorial_last_screen_text),
-              highlightArea = { FullScreenSemiTransparentOverlay() },
-              onNext = {
-                navigationActions.navigateTo(Screen.HOME)
-                onFinish()
-              },
-              textTag = "CompletionTextTag")
+    // Add the blocking overlay
+    BlockingInteractionsOverlay()
+
+    // Render the current tutorial step
+    if (step in tutorialSteps.indices) {
+      val isLastStep = step == tutorialSteps.size - 1
+      val currentStep = tutorialSteps[step]
+      TutorialOverlay(
+          text = currentStep.text,
+          highlightArea = currentStep.highlightArea,
+          onNext = {
+            if (isLastStep) {
+              navigationActions.navigateTo(Screen.HOME)
+              onFinish()
+            } else {
+              step++
+            }
+          },
+          textTag = currentStep.textTag)
     }
+
+    SkipButton(
+        onSkip = {
+          navigationActions.navigateTo(Screen.HOME)
+          onFinish()
+        })
```

Comment:
Adding the 'Skip' button is a thoughtful addition that enhances user experience by giving users control over the tutorial flow.